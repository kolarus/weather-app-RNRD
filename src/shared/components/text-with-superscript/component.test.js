import React from 'react';
import {render} from '@testing-library/react-native';

import TextWithSuperscript from './component';

test('TextWithSuperscript should match snapshot', () => {
  const {toJSON} = render(
    <TextWithSuperscript fontSize={10} superScript="o">
      2
    </TextWithSuperscript>,
  );

  expect(toJSON()).toMatchSnapshot();
});

test('TextWithSuperscript should render passed text', () => {
  const {getByText} = render(
    <TextWithSuperscript fontSize={10} superScript="o">
      2
    </TextWithSuperscript>,
  );

  expect(getByText('2')).not.toBeNull();
  expect(getByText('o')).not.toBeNull();
});

test('CommonText should render passed text with correct font size', () => {
  const {getByText} = render(
    <TextWithSuperscript fontSize={10} superScript="o">
      2
    </TextWithSuperscript>,
  );

  expect(getByText('2')).toHaveStyle({fontSize: 10});
});
