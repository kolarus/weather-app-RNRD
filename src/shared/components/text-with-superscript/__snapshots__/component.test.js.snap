// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextWithSuperscript should match snapshot 1`] = `
<View
  style={
    Array [
      Object {
        "flexDirection": "row",
      },
      undefined,
    ]
  }
>
  <Text
    style={
      Array [
        Object {
          "color": "white",
          "fontFamily": "PT Sans",
          "fontSize": 15,
        },
        Array [
          undefined,
          Object {
            "fontSize": 10,
          },
        ],
        Object {},
      ]
    }
  >
    2
  </Text>
  <Text
    style={
      Array [
        Object {
          "color": "white",
          "fontFamily": "PT Sans",
          "fontSize": 15,
        },
        Array [
          Object {
            "fontSize": 6,
            "lineHeight": 7.199999999999999,
          },
          undefined,
        ],
        Object {},
      ]
    }
  >
    o
  </Text>
</View>
`;
